### ISSUE TITLE
raise NotImplementedError("By default it is assumed that the data loader used for initialize "

### ISSUE THREAD

**Issue by SiChuanJay** [openvinotoolkit/openvino_notebooks#868]
Hello, I ran into a problem with the training-time quantization of yolov8 using nncf, as shown below. I refer to the example link (https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/302-pytorch-quantization-aware-training). My v8_dataloader is also of type DataLoader, but I got an error. Thanks very much!

![图片](https://user-images.githubusercontent.com/52130681/221780915-e5006165-6a1e-4029-8ee2-9a988cc7c4a8.png)
![图片](https://user-images.githubusercontent.com/52130681/221780923-c42d141e-8ab5-495b-a706-de8c121d107a.png)

**Comment by MaximProshin** [comment]
@SiChuanJay , thanks for your interest! In general, we expect you to raise issues about nncf to nncf repo (https://github.com/openvinotoolkit/nncf), not here. 

As far as I can see you're applying training-aware quantization for yolov8 while it can be successfully quantized in post-training. Can you please take a look at https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/openvino/yolov8 or https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/230-yolov8-optimization?

Regarding your error, I think it's due to the way you've implemented the dataloader. I let @vshampor provide more thoughts about it and how it could be fixed on your side.

**Comment by SiChuanJay** [comment]
Yes, I have tried the post-training method and it can be successful. But the train-time approach will fine-tune the model after it is converted to int8, which might work better, so I wanted to give it a try. Besides, I have another question. The int8 model I got using the example you provided is the same size as the fp32 model. Why?
![图片](https://user-images.githubusercontent.com/52130681/221850913-777a2a80-cded-445b-834e-702750da6190.png)
